Dockerfile

```diff
@@ -1,18 +1,18 @@
 ############# builder
-FROM golang:1.17.5 AS builder
+FROM golang:1.18.1 AS builder
```

Comment:
Can you also update `go.mod` please and prepare the update for the `meta/ci` ref with the pipeline definition?

Reply:
Because the pipeline image is not updated, the CI job fails with 

>go mod tidy: go.mod file indicates go 1.18, but maximum supported version is 1.17

Reply:
Also seeing some  other errors in CI  --> I revendored after the go.mod update, and many dependencies got updated 
@rfranzke have you seen this permission issue with vendored gardener in other extensions already? 

> make[1]: /go/src/github.com/gardener/gardener-extension-runtime-gvisor/vendor/github.com/gardener/gardener/hack/get-build-ld-flags.sh: Permission denied
make[1]: /go/src/github.com/gardener/gardener-extension-runtime-gvisor/vendor/github.com/gardener/gardener/hack/clean.sh: Permission denied

Reply:
> @rfranzke have you seen this permission issue with vendored gardener in other extensions already?

Do you use `make revendor`? ref https://github.com/gardener/gardener-extension-runtime-gvisor/blob/3ff11648df922a5c02dccd3984d5267335a5b227/Makefile#L69-L70

Reply:
> it seems like I do not have to access rights anymore.

I'm taking care.

Reply:
@danielfoehrKn I've updated the pipeline to go@v1.18, however it's still failing, please check

Reply:
Waiting for [this PR](https://github.com/gardener/gardener/pull/5896) to go through so that g/g can vendored with a golangci-lint version that supports `1.18.1`  without generics. Otherwise the CI `make check`  won't work

Reply:
Hm, OK, I wasn't aware. In this case, let's proceed with 1.17.9 for now since we don't want to wait for new g/g release + revendoring. I'm changing the pipeline back to 1.17, please adapt the PR.

Reply:
okay, golang the image with 1.17.9 also has no reported vulnerabilities

![image](https://user-images.githubusercontent.com/33809186/167104091-8c648634-31bb-488c-9efb-dfe0b3341708.png)